lib/src/builder/parsing/members_from_ast.dart

```diff
@@ -429,26 +426,54 @@ class _BoilerplateMemberDetector {
     return false;
   }
 
+  /// UiComponent, UiComponent2, UiStatefulComponent, FluxUiComponent, CustomUiComponent, ...
+  static final _componentBaseClassPattern = RegExp('Ui\w+Component');
```

Comment:
This doesn't actually match `FluxUiComponent` or `CustomUiComponent`, which might be fine but should update the doc comment.

```suggestion
  static final _componentBaseClassPattern = RegExp('.*Ui\w+Component');
```

Reply:
Good catch!  I think it actually needs to be
```suggestion
  static final _componentBaseClassPattern = RegExp(r'Ui\w*Component');
```
though.

Also FYI the regex matches even if it's a partial match:
```dart
main() {
  final _componentBaseClassPattern = RegExp(r'Ui\w*Component');
  const names = [
    'UiComponent', 
    'UiComponent2', 
    'UiStatefulComponent',
    'FluxUiComponent', 
    'CustomUiComponent',
  ];
  
  for (var name in names) {
    final hasMatch = _componentBaseClassPattern.hasMatch(name);
    print('hasMatch for "$name": $hasMatch');
  }
}
```
```
hasMatch for "UiComponent": true
hasMatch for "UiComponent2": true
hasMatch for "UiStatefulComponent": true
hasMatch for "FluxUiComponent": true
hasMatch for "CustomUiComponent": true
```